src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp

```diff
@@ -46,16 +46,12 @@ class SerialBlockOffsetSharedArray: public CHeapObj<mtGC> {
   // Array for keeping offsets for retrieving object start fast given an
   // address.
   VirtualSpace _vs;
-  uint8_t* _offset_array;          // byte array keeping backwards offsets
 
-  void fill_range(size_t start, size_t num_cards, uint8_t offset) {
-    void* start_ptr = &_offset_array[start];
-    memset(start_ptr, offset, num_cards);
-  }
+  // Biased array-start of BOT array for fast BOT entry translation
+  uint8_t* _offset_base;
 
-  uint8_t offset_array(size_t index) const {
-    assert(index < _vs.committed_size(), "index out of range");
-    return _offset_array[index];
+  void fill_range(const uint8_t* const start, size_t num_cards, uint8_t offset) {
```

Comment:
I think the first arg should be `uint8_t* const start`. Then, the `void*` cast below can be dropped.

Reply:
Fixed.